arch_prctl.c

```diff
@@ -122,7 +122,7 @@ UK_LLSYSCALL_R_DEFINE(long, arch_prctl, long, code, long, addr, long, arg2)
 		case ARCH_SET_FS:
 			uk_pr_debug("arch_prctl option SET_FS(%p)\n",
 				    (void *) addr);
-			writefs((__uptr) addr);
+			_uk_syscall_ultlsp = (__uptr)addr;
```

Comment:
As indicated by the leading underscore, `_uk_syscall_ultlsp` should be a variable private to the `libsyscall_shim`. The only reason this works is because `libsyscall_shim` does not define `exportsyms.uk`. 

Perhaps it would be cleaner to add a setter function in https://github.com/unikraft/unikraft/pull/1173 and call that from here. Saying that, we already have `ukplat_tlsp_get/set` API which may make things confusing, so the name of this accessor will have to be selected carefully. Perhaps `uk_syscall_ultlsp_get/set()`?

Reply:
Oh yes it makes sense! I did not give much thought to this PR or the #1173 one since they are overwritten anyway. Besides [this](https://github.com/unikraft/unikraft/pull/1175/commits/b43d2c53d846854bf4016a6f94519dbc9f6049d5#diff-db42214216f40633c2a5e6d355fb0a0799882966960f09333520660fef1dab27L188) will delete that variable anyway. But yes, I will add a setter (the getter is already there). Thanks for the express review! :heart:

Reply:
Following an offline discussion and since these changes will be overwritten by #65 and [unikraft #1173](https://github.com/unikraft/unikraft/pull/1173) anyway (this commit is just transitory and simply meant to be cherry-pickable and present in the git history to make people aware that there has been a bug), we decided not to bother with a `setter/getter` and proceed further.